src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -337,6 +427,12 @@ public boolean setHeaders(HttpURLConnection conn, HeaderParser p, String raw) {
         if (algorithm == null || algorithm.isEmpty()) {
             algorithm = "MD5";  // The default, accoriding to rfc2069
         }
+        var oid = KnownOIDs.findMatch(algorithm.toUpperCase(Locale.ROOT));
```

Comment:
No need to call `toUpperCase`. `findMatch` already called that. On the other hand, we don't support the name `SHA-512-256`, and if you translate it to stdName here there is no need to do it again in `computeUserhash`.